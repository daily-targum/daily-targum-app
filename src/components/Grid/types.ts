import React from 'react';
import { ViewStyle, TextStyle, ImageStyle } from 'react-native';

// type definitions
export type NativeStyle = ViewStyle | TextStyle | ImageStyle;

export interface GlobalBreakPoints {
  xs: number,
  sm: number,
  md: number,
  lg: number,
  xl: number,
  xxl: number
}

export interface BreakPoints {
  xs?: number | string,
  sm?: number | string,
  md?: number | string,
  lg?: number | string,
  xl?: number | string,
  xxl?: number | string
}

export interface ComputedBreakPoints {
  xs: number | string,
  sm: number | string,
  md: number | string,
  lg: number | string,
  xl: number | string,
  xxl: number | string
}

export interface ComputedStyles {
  xs?: NativeStyle
  sm?: NativeStyle
  md?: NativeStyle
  lg?: NativeStyle
  xl?: NativeStyle
  xxl?: NativeStyle
}

export type BreakPoint = 'xs' | 'sm' | 'md' | 'lg' | 'xl' | 'xxl';

export interface ColProps extends BreakPoints {
  style?: NativeStyle
  children?: React.ReactNode
}

export interface RowProps extends BreakPoints {
  style?: NativeStyle
  children?: React.ReactNode
  spacing?: number
}

export interface ConsumerProps {
  children: React.ReactElement
}