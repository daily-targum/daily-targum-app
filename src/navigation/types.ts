import { RouteProp } from '@react-navigation/core';

export type RootStackParamList = {
  Page: { slug: string },
  ArticleCategory: { category: string }
};

export type PageRouteProp = RouteProp<RootStackParamList, 'Page'>;
export type ArticleCategoryPageProp = RouteProp<RootStackParamList, 'ArticleCategory'>;