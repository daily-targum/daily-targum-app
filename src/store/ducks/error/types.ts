const DISPLAY_ERROR = 'DISPLAY_ERROR';
const CLEAR_ERROR = 'CLEAR_ERROR';

export default {
  DISPLAY_ERROR,
  CLEAR_ERROR
}

export interface State {
  error: string | null
}