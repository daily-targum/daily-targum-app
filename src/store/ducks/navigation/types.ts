export const types = {
  SET_TAB_BAR_VISIBLE: 'SET_TAB_BAR_VISIBLE'
};

export interface State {
  bottomTabBarVisible: boolean
}