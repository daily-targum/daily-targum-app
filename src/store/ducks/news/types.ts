import { Article } from '../../../shared/src/client';

export const types = {
  CATEGORY_REFRESHING: 'CATEGORY_REFRESHING',
  CATEGORY_SET: 'CATEGORY_SET',
  SET_REFRESH_ALL: 'SET_REFRESH_ALL',
  SAVE_SCROLL_INDEX: 'SAVE_SCROLL_INDEX'
};

export interface State {
  feed: {
    [key in string]: {
      data: Article[],
      refreshing: boolean,
      nextToken?: string | null,
      scrollIndex: 0
    }
  },
  refreshingAll: boolean,
  loading: boolean
}