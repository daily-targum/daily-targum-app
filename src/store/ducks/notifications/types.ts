const TOKEN_SET = 'TOKEN_SET';
const HAS_PERMISSION_SET = 'HAS_PERMISSION_SET';
const ENABLED_SET = 'ENABLED_SET';
const AUTO_PROMPT_DISABLE = 'AUTO_PROMPT_DISABLE';

export default {
  TOKEN_SET,
  HAS_PERMISSION_SET,
  ENABLED_SET,
  AUTO_PROMPT_DISABLE
};

export interface State {
  token: string,
  hasPermission: boolean | null,
  enabled: boolean,
  autoPrompt: boolean
}