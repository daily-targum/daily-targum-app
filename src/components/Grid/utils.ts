import { breakPointKeys, breakPoints } from './config';
import { BreakPoint, BreakPoints, ComputedBreakPoints } from './types';

// Web and Native
export function getBreakpoint(windowWidth: number): BreakPoint {
  let breaker: BreakPoint = 'xs';
  breakPointKeys.forEach((key: BreakPoint) => {
    if(windowWidth > breakPoints[key]) {
      breaker = key;
    }
  });
  return breaker;
}

// Web and Native
export function computeBreakpoints(breakpoints: BreakPoints): ComputedBreakPoints {
  let computedBreakPoints: BreakPoints = {};
  let crntWidth = '';
  breakPointKeys.forEach(key => {
    if(typeof breakpoints[key] === 'number') {
      // @ts-ignore
      crntWidth = breakpoints[key] / 0.24 + '%';
    } else if(typeof breakpoints[key] === 'string') {
      // @ts-ignore
      crntWidth = breakpoints[key];
    }
    computedBreakPoints[key] = crntWidth;
  });
  return computedBreakPoints as any;
}