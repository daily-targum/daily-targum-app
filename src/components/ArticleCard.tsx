import React from 'react';
import Theme from '../components/Theme';
import { View, Image, TouchableOpacity } from 'react-native';
import { Text } from 'react-native-paper';
import { GetArticle } from '../shared/src/client';
import { useNavigation } from '@react-navigation/core';
import { formatDateAbriviated } from '../shared/src/utils';
import { shareArticle } from '../utils';
import { LinearGradient } from 'expo-linear-gradient';

function Large({
  article,
  width,
  style
}: {
  article: GetArticle,
  width: number | string,
  style?: any
}) {
  const styles = Theme.useStyleCreator(styleCreator);
  const navigation = useNavigation();
  function handlePress() {
    navigation.navigate('Article', {
      // id: '6jd2Gmkw7kuC3GWnjiuYY7',
      id: article.id
    });
  }
  return (
    <TouchableOpacity 
      onPress={handlePress}
      onLongPress={() => shareArticle({article})}
      activeOpacity={0.7}
      style={[{width}, style]}
    >
      <View style={styles.cardLarge}>
        <Image 
          source={{uri: article.media+'?h=260&w=400&fit=crop&crop=faces,center'}}
          style={styles.cardLargeImage}
          resizeMethod="resize"
        />
        <LinearGradient
          colors={['transparent', '#222']}
          style={{
            position: 'absolute',
            height: '100%',
            width: '100%'
          }}
        />
        <View style={styles.cardLargeContent}>
          <Text numberOfLines={1} style={styles.cardSubtitleLarge}>{formatDateAbriviated(article.publishDate)}</Text>
          <Text numberOfLines={2} style={styles.cardTitleLarge}>{article.title}</Text>
        </View>
      </View>
    </TouchableOpacity>
  );
}

function Medium({
  article,
  style
}: {
  article: GetArticle,
  style?: any
}) {
  const styles = Theme.useStyleCreator(styleCreator);
  const navigation = useNavigation();
  function handlePress() {
    navigation.navigate('Article', {
      // id: '6jd2Gmkw7kuC3GWnjiuYY7',
      id: article.id
    });
  }
  return (
    <TouchableOpacity 
      onPress={handlePress}
      onLongPress={() => shareArticle({article})}
      activeOpacity={0.7}
      style={style}
    >
      <View style={styles.cardMedium}>
        {article.media ? (
          <Image 
            source={{uri: article.media+'?h=260&w=400&fit=crop&crop=faces,center'}}
            style={styles.cardMediumImage}
            resizeMethod="resize"
          />
        ) : null}
        <View style={styles.cardMediumContent}>
          <Text numberOfLines={1} style={styles.cardSubtitle}>{formatDateAbriviated(article.publishDate)}</Text>
          <Text numberOfLines={3} style={styles.cardTitle}>{article.title}</Text>
        </View>
      </View>
    </TouchableOpacity>
  );
}

function Small({
  article,
  width,
  style
}: {
  article: GetArticle,
  width: number | string,
  style?: any
}) {
  const styles = Theme.useStyleCreator(styleCreator);
  const navigation = useNavigation();
  function handlePress() {
    navigation.navigate('Article', {
      // id: '6jd2Gmkw7kuC3GWnjiuYY7',
      id: article.id,
      article
    });
  }
  return (
    <TouchableOpacity 
      onPress={handlePress}
      onLongPress={() => shareArticle({article})}
      activeOpacity={0.7}
      style={[{width}, style]}
    >
      <View style={styles.cardSmall}>
        {article.media ? (
          <Image 
            source={{uri: article.media+'?h=200&w=300&fit=crop&crop=faces,center'}}
            style={styles.cardSmallImage}
            resizeMethod="resize"
          />
        ) : null}
        <View style={styles.cardSmallContent}>
          <Text numberOfLines={1} style={styles.cardSubtitle}>{formatDateAbriviated(article.publishDate)}</Text>
          <Text numberOfLines={3} style={styles.cardTitle}>{article.title}</Text>
        </View>
      </View>
    </TouchableOpacity>
  );
}

const styleCreator = Theme.makeStyleCreator(theme => ({
  // Large Card
  cardLarge: {
    flexDirection: 'column',
    borderRadius: theme.roundness(1),
    overflow: 'hidden',
    marginBottom: theme.spacing(1.5)
  },
  cardLargeImage: {
    width: '100%',
    aspectRatio: 3/2
  },
  cardLargeContent: {
    position: 'absolute',
    bottom: 0,
    flex: 1,
    padding: theme.spacing(2)
  },
  cardTitleLarge: {
    fontSize: 20,
    fontWeight: '700',
    color: '#fff'
  },
  cardSubtitleLarge: {
    fontSize: 14,
    color: theme.colors.textMuted,
    paddingBottom: theme.spacing(1)
  },
  // Medium Card
  cardMedium: {
    flexDirection: 'column',
    width: 190
  },
  cardMediumImage: {
    width: '100%',
    borderRadius: theme.roundness(1),
    aspectRatio: 3/2
  },
  cardMediumContent: {
    flex: 1,
    padding: theme.spacing(1.5)
  },
  // Small Card
  cardSmall: {
    flexDirection: 'row',
    marginBottom: theme.spacing(1.5),
    overflow: 'hidden'
  },
  cardSmallImage: {
    flex: 1,
    width: '40%',
    maxWidth: '40%',
    borderRadius: theme.roundness(1),
    aspectRatio: 3/2
  },
  cardSmallContent: {
    flex: 1,
    paddingTop: theme.spacing(1),
    paddingLeft: theme.spacing(1.5),
    paddingRight: 0,
    paddingBottom: 0
  },
  // All Cards
  cardTitle: {
    fontSize: 14,
    fontWeight: '700'
  },
  cardSubtitle: {
    fontSize: 14,
    color: theme.colors.textMuted,
    paddingBottom: theme.spacing(1)
  }
}));

export default {
  Large,
  Medium,
  Small
};