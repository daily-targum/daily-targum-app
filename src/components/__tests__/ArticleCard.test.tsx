jest.mock('@react-navigation/core', () => {
  const navigate = jest.fn();
  return {
    useNavigation: () => ({
      navigate
    })
  };
});

jest.mock('../Theme');

import React from 'react';
import { Article } from '../../types';
import ArticleCard from '../ArticleCard';
import rederer from 'react-test-renderer';
import { useNavigation } from '@react-navigation/core'
import { TouchableOpacity } from 'react-native';
import * as share from '../../utils/share';
jest.spyOn(share, 'shareArticle');

const ARTICLE: Article = {
  id: '982-af8ojpjoiadfp83498',
  title: 'About',
  media: 'https://google.com',
  author: 'Jon Doe',
  date: '01-18-2020',
  content: '',
  url: 'https://google.com'
};

describe('<ArticleCard />', () => {

  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('small card renders correctly', () => {
    const tree = rederer.create(
      <ArticleCard.Small article={ARTICLE} width='100%'/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('small card handle press', () => {
    const tree = rederer.create(
      <ArticleCard.Small article={ARTICLE} width='100%'/>
    );
    tree.root.findByType(TouchableOpacity).props.onPress();
    expect(useNavigation().navigate).toBeCalledTimes(1);
  });

  it('small card handle long press', () => {
    const tree = rederer.create(
      <ArticleCard.Small article={ARTICLE} width='100%'/>
    );
    tree.root.findByType(TouchableOpacity).props.onLongPress();
    expect(share.shareArticle).toBeCalledTimes(1);
  });

  it('medium card renders correctly', () => {
    const tree = rederer.create(
      <ArticleCard.Medium article={ARTICLE}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('medium card handle press', () => {
    const tree = rederer.create(
      <ArticleCard.Medium article={ARTICLE}/>
    );
    tree.root.findByType(TouchableOpacity).props.onPress();
    expect(useNavigation().navigate).toBeCalledTimes(1);
  });

  it('medium card handle long press', () => {
    const tree = rederer.create(
      <ArticleCard.Medium article={ARTICLE}/>
    );
    tree.root.findByType(TouchableOpacity).props.onLongPress();
    expect(share.shareArticle).toBeCalledTimes(1);
  });
});