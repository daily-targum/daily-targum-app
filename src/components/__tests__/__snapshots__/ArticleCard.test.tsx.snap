// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleCard /> medium card renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "column",
        "width": 190,
      }
    }
  >
    <Image
      resizeMethod="resize"
      source={
        Object {
          "uri": "https://snworksceo.imgix.net/rdt/bb7e35ca-1fbf-4b4e-8b9b-c4daa6d0ce33.sized-1000x1000.JPG?h=260&w=400&fit=crop&crop=faces,center",
        }
      }
      style={
        Object {
          "aspectRatio": 1.5,
          "borderRadius": 10,
          "width": "100%",
        }
      }
    />
    <View
      style={
        Object {
          "flex": 1,
          "padding": 10.5,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000",
            },
            null,
            Object {
              "color": "#999",
              "fontSize": 14,
              "paddingBottom": 7,
            },
          ]
        }
      >
        May 4
      </Text>
      <Text
        numberOfLines={3}
        style={
          Array [
            Object {
              "color": "#000",
            },
            null,
            Object {
              "fontSize": 14,
              "fontWeight": "700",
            },
          ]
        }
      >
        Rutgers' NCAA Tournament hopes depend on team-wide performance
      </Text>
    </View>
  </View>
</View>
`;

exports[`<ArticleCard /> small card renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 10.5,
        "overflow": "hidden",
      }
    }
  >
    <Image
      resizeMethod="resize"
      source={
        Object {
          "uri": "https://snworksceo.imgix.net/rdt/bb7e35ca-1fbf-4b4e-8b9b-c4daa6d0ce33.sized-1000x1000.JPG?h=200&w=300&fit=crop&crop=faces,center",
        }
      }
      style={
        Object {
          "aspectRatio": 1.5,
          "borderRadius": 10,
          "flex": 1,
          "maxWidth": "40%",
          "width": "40%",
        }
      }
    />
    <View
      style={
        Object {
          "flex": 1,
          "paddingBottom": 0,
          "paddingLeft": 10.5,
          "paddingRight": 0,
          "paddingTop": 7,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000",
            },
            null,
            Object {
              "color": "#999",
              "fontSize": 14,
              "paddingBottom": 7,
            },
          ]
        }
      >
        May 4
      </Text>
      <Text
        numberOfLines={3}
        style={
          Array [
            Object {
              "color": "#000",
            },
            null,
            Object {
              "fontSize": 14,
              "fontWeight": "700",
            },
          ]
        }
      >
        Rutgers' NCAA Tournament hopes depend on team-wide performance
      </Text>
    </View>
  </View>
</View>
`;
